### ISSUE TITLE
[Bug]: aws_ebs_volume tags attribute is empty map of strings

### ISSUE THREAD

**Issue by antm-pp** [hashicorp/terraform-provider-aws#37736]
### Terraform Core Version

1.8.4

### AWS Provider Version

5.49.0, 5.50.0, 5.51.0, 5.51.1

### Affected Resource(s)

* aws_ebs_volume

### Expected Behavior

In 5.48 our existing config works fine.  

data.aws_ebs_volume.asm["<instance-id>"] is used in a for loop to create a local variable that contains the contents of a particular tag:

asm_data_devices = [for v in data.aws_ebs_volume.asm : v.tags["Device"] if v.tags["Function"] == "ASM:DATA"]

This should create a list of device ids.

### Actual Behavior

For each iteration of the for loop:

Error: Invalid index, v.tags is empty map of string.  So somehow the for loop can't read the tags attribute of the data source.

### Relevant Error/Panic Output Snippet

```shell
Error: Invalid index
│ 
│   on storage.tf line 65, in locals:
│   65:   asm_data_devices = [for v in data.aws_ebs_volume.asm : v.tags["Device"] if v.tags["Function"] == "ASM:DATA"]
│     ├────────────────
│     │ v.tags is empty map of string
│ 
│ The given key does not identify an element in this collection value.
```

### Terraform Configuration Files

Working constraints prevent further sharing of data

### Steps to Reproduce

Working constraints prevent further sharing of data

### Debug Output

Working constraints prevent further sharing of data

### Important Factoids

Appreciate I haven't provided much context for the error, but it seems quite specific to a resource that should be producing an output and isn't. The physical AWS resources are constant ie have tags.

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @antm-pp 👋 Thank you for taking the time to raise this! Can you inspect what `data.aws_ebs_volume.asm.tags` looks like during the plan? Does that change when reverting to version `5.48`?

If you're able to supply debug logging, that may also be extra helpful to whoever picks this up, since this is likely very resource-specific.

**Comment by justinretzolk** [comment]
Since we haven't heard back, I'm going to close this issue. If you're still having trouble, please feel free to open a new issue, referencing this one for context as needed.

For what it's worth, it looks like this may have been fixed in #37966, which was included in `5.54.0`.
